### DISCUSSION TITLE
How to serach \n

### DISCUSSION THREAD

**Discussion by kathimohan** [jsvine/pdfplumber#853] [Q&A]
Not able to find new line using search \n, Is there any way to find new new line and serach for \n

pattern_n = page.search(r"\n")
*** ValueError: min() arg is an empty sequence
pattern_n = page.search(r"\s")
*** ValueError: min() arg is an empty sequence

**Comment by jsvine** [comment]
Hi @kathimohan. This is a really interesting question, and thank you for identifying this bug. I'm working on a fix for the error, which occurs because there is no internal PDF representation of a "newline"; we just perceive newlines based on the distances between lines of text. I spent some time today trying to derive the implicit "position" of the newlines, but nothing felt quite right. So, for now, searching for newlines will just return a blank result set.

I'm also working on some functionality related to your question here, which will provide a more convenient way to identify where newlines are: https://github.com/jsvine/pdfplumber/discussions/852

**Reply by jsvine** [reply]
As of `v0.9.0`, `pdfplumber` now returns an empty list (instead of throwing an error) when searching for whitespace or newline-only patterns.

**Reply by kathimohan** [reply]
Thank you very much for your help, but for now fixed my requirement by changing the logic, will check based on need and will reply
